runtime/commonMain/src/kotlinx/serialization/Descriptors.kt

```diff
@@ -11,6 +11,12 @@ sealed class SerialKind {
 
 sealed class PrimitiveKind : SerialKind() {
     object INT : PrimitiveKind()
+
+    @Deprecated(
+        message = "Primitive kind Unit is deprecated with no replacement. Use UnionKind.OBJECT instead",
```

Comment:
That's not very true, because every Unit is an OBJECT, but not every OBJECT is a Unit.

Reply:
What exactly is not true?
Deprecation suggests replacing `PrimitiveKind.Unit` with `Object`, not the vice versa

Reply:
If I had a branch `is PrimitiveKind.Unit` and I replace it with `UnionKind.OBJECT`, now other objects fell down into this branch, not only `kotlin.Unit` ones.

Reply:
Good point, then I'll leave clarified text and remove replacement completely